### PR TITLE
Added ipv6 resolve support to webrtc resolving

### PR DESCRIPTION
**Pull request by bendpx**

### CODE REVIEW

**Comment by bendpx** [comment]
Uncaught TypeError: Cannot read property '1' of null
n.Mixin.extend.findUserIPs.i.onicecandidate

Chrome Browser Version 43.0.2357.81 (64-bit):
"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/43.0.2357.81 Safari/537.36"

Insight:
c.exec(e.candidate.candidate)[1];

Where:
c = /([0-9]{1,3}(\.[0-9]{1,3}){3})/
e.candidate.candidate = "candidate:555245142 2 udp 2122197246 2601:645:8000:3c81:640c:1fb3:db7b:cb94 55314 typ host generation 0"

and c.exec(e.candidate.candidate) returns null

File: index.html [hunk 1]
```diff
@@ -62,25 +62,28 @@
...
-                function handleCandidate(candidate){
+                function handleCandidate(candidate) {
                     //match just the IP address
-                    var ip_regex = /([0-9]{1,3}(\.[0-9]{1,3}){3})/
-                    var ip_addr = ip_regex.exec(candidate)[1];
+                    var ipv4_regex = /([0-9]{1,3}(\.[0-9]{1,3}){3})/;
+                    var ipv6_regex = /(([0-9a-fA-F]{1,4}:){7,7}[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,7}:|([0-9a-fA-F]{1,4}:){1,6}:[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,5}(:[0-9a-fA-F]{1,4}){1,2}|([0-9a-fA-F]{1,4}:){1,4}(:[0-9a-fA-F]{1,4}){1,3}|([0-9a-fA-F]{1,4}:){1,3}(:[0-9a-fA-F]{1,4}){1,4}|([0-9a-fA-F]{1,4}:){1,2}(:[0-9a-fA-F]{1,4}){1,5}|[0-9a-fA-F]{1,4}:((:[0-9a-fA-F]{1,4}){1,6})|:((:[0-9a-fA-F]{1,4}){1,7}|:)|fe80:(:[0-9a-fA-F]{0,4}){0,4}%[0-9a-zA-Z]{1,}|::(ffff(:0{1,4}){0,1}:){0,1}((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])|([0-9a-fA-F]{1,4}:){1,4}:((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9]))/;
+                    var ip_addr = ipv4_regex.exec(candidate);
+
+                    if (!ip_addr || typeof ip_addr == 'undefined') {
```
Comment (diafygi) [inline index.html]: Please add a comment explaining this.

File: index.html [hunk 2]
```diff
@@ -62,25 +62,28 @@
...
+                    if (ip_addr && typeof ip_addr != 'undefined') {
+                        ip_addr = ip_addr[1];
+                    } else {
+                        return callback("");
+                    }
 
                     //remove duplicates
-                    if(ip_dups[ip_addr] === undefined)
-                        callback(ip_addr);
+                    if (ip_dups[ip_addr] === undefined)
```
Comment (diafygi) [inline index.html]: Why was this changed?

File: index.html [hunk 3]
```diff
@@ -62,25 +62,28 @@
...
+                        return callback(ip_addr);
 
                     ip_dups[ip_addr] = true;
                 }
 
-                //listen for candidate events
```
Comment (diafygi) [inline index.html]: Why was this removed?
